tests/unit/htdocs/class/XoopsUserUtilityTest.php

```
@@ -0,0 +1,508 @@
+     * @param string $ip       The IP address to set as REMOTE_ADDR
+     * @param int    $expected The expected ip2long() value
+     */
+    #[\PHPUnit\Framework\Attributes\DataProvider('ipAddressProvider')]
+    public function testVariousIpAddressesReturnCorrectIp2long(string $ip, int $expected): void
+    {
+        $_SERVER['REMOTE_ADDR'] = $ip;
+
+        $result = \XoopsUserUtility::getIP(false);
+
+        $this->assertSame($expected, $result);
+    }
```

PHPMD (2.15.0)

[error] testVariousIpAddressesReturnCorrectIp2long accesses the super-global variable $_SERVER. (undefined)

(Superglobals)